### PR TITLE
Require at least 1 child in JSX components

### PR DESCRIPTION
**Pull request by Mrtenz**
This changes the types and validation for JSX components to require at least one child. Previously it was possible to do something like `<Box children={[]} />`, but this will now result in an error.

### CODE REVIEW

File: packages/snaps-sdk/src/jsx/validation.ts [hunk 1]
```diff
@@ -76,6 +78,22 @@ export const ElementStruct: Describe<GenericSnapElement> = object({
...
+ * @returns The struct for the non-empty array type.
+ */
+function nonEmptyArray<Type, Schema>(
+  struct: Struct<Type, Schema>,
+): Struct<NonEmptyArray<Type>, any> {
+  return refine(
+    array(struct),
+    'NonEmptyArray',
+    (value) => value.length > 0,
+  ) as unknown as Struct<NonEmptyArray<Type>, Schema>;
```
Comment (FrederikBolding) [inline packages/snaps-sdk/src/jsx/validation.ts:94]: Could we just use `nonempty` from superstruct?

File: packages/snaps-sdk/src/jsx/validation.test.tsx [hunk 2]
```diff
@@ -430,6 +430,8 @@ describe('BoxStruct', () => {
     [],
     // @ts-expect-error - Invalid props.
     <Box />,
+    // @ts-expect-error - Invalid props.
+    <Box children={[]} />,
```
Comment (FrederikBolding) [inline packages/snaps-sdk/src/jsx/validation.test.tsx:434]: Can we assert that empty `Dropdown` is invalid too?

File: packages/snaps-sdk/src/jsx/validation.ts (hunk 1 above)
Comment (Mrtenz) [reply packages/snaps-sdk/src/jsx/validation.ts]: Yes, I didn't know that was a thing.

**Review by FrederikBolding — APPROVED** [review]
LGTM but there are some type problems in CI it seems

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**